File: dotnet/cafdemalihapa/Directories/WorkingCirculationDirectory.cs
namespace cafdemalihapa.Directories
{
    public class WorkingCirculationDirectory
    {
        public string GetName()
        {
            var workingCirculationDirectoryName = GetPath();

            var environmentVariablesManagementDirectoryName =
                Path.Combine(workingCirculationDirectoryName, "EnvironmentVariablesManagement");

            return environmentVariablesManagementDirectoryName;
        }

        private string GetPath()
        {
            const string repositoryDirectoryNameKey = "--repository-directory";
            var workingCirculationDirectoryName =
                Path.Combine(CommandLineArgs.GetByKey(repositoryDirectoryNameKey), "WorkingCirculation");
            return workingCirculationDirectoryName;
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Directories/WorkingCirculationDirectory.fs
namespace cafdemwimu.console.Directories

open System.IO
open cafdemwimu.console

type WorkingCirculationDirectory() =
    member private _.GetPath() =
        let repositoryDirectoryNameKey = "--repository-directory"
        let workingCirculationDirectoryName =
            Path.Combine(CommandLineArgs.GetByKey(repositoryDirectoryNameKey), "WorkingCirculation")
        workingCirculationDirectoryName
